ably/proto_message_decoding_test.go

```diff
@@ -0,0 +1,147 @@
+package ably
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ably/ably-go/ably/internal/ablyutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
```

Comment:
Should we use either `assert` or `require` rather than a mixture of both?

Reply:
I am all for reducing dependencies. 
But assert and require do different things.
We use assert to record an error, but allow the test to continue to run, so other failures can be found.
On the other hand use require for things like failure to open the fixture file, where there is no way to continue the error and we need to abort the test now.

Reply:
Ok thanks, so let me instead comment in the code, since on a first pass I was under the impression this was not how they were being used and you meant to use `require`.